src/Support/InferExtensions/ResponderTypeInfer.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace Dedoc\Scramble\Support\InferExtensions;
+
+use Dedoc\Scramble\Infer\Definition\ClassDefinition;
+use Dedoc\Scramble\Infer\Extensions\ExpressionTypeInferExtension;
+use Dedoc\Scramble\Infer\Scope\Scope;
+use Dedoc\Scramble\Support\Type\ObjectType;
+use Dedoc\Scramble\Support\Type\Type;
+use League\Fractal\TransformerAbstract;
+use PhpParser\Node;
+use PhpParser\Node\Expr;
+use Dedoc\Scramble\Support\Generator\Types\UnknownType;
+use Dedoc\Scramble\Infer\Services\FileNameResolver;
+use Dedoc\Scramble\Support\ResponseExtractor\ModelInfo;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Str;
+
+class ResponderTypeInfer implements ExpressionTypeInferExtension
+{
+    private $modelType;
+    private string $modelPropertyName;
+    public static $transformerModelTypesCache = [];
+
+    public function getType(Expr $node, Scope $scope): ?Type
+    {
+        if (
+            !$scope->classDefinition()?->isInstanceOf(TransformerAbstract::class)
+            || $scope->classDefinition()?->name === TransformerAbstract::class
+        ) {
+            return null;
+        }
+
+        $transformerArguments = $scope->classDefinition()->getMethodDefinition('transform')->type->arguments;
+
+        $this->modelPropertyName = array_key_first($transformerArguments);
+        $this->modelType = $transformerArguments[$this->modelPropertyName]->is;
+
+        /** $this->? */
+        if (
+            $node instanceof Node\Expr\PropertyFetch && $node->var?->name === $this->modelPropertyName
+            && is_string($node->name->name ?? null)
+            && !array_key_exists($node->name->name, $scope->classDefinition()->properties)
+            && ($type = $this->modelType($scope->classDefinition(), $scope))
+        ) {
+            return $scope->getPropertyFetchType($type, $node->name->name);
+        }
+
+        /**
+         * new MissingValue
+         */
```

Comment:
What is this for?

Reply:
The comment? To make reference the type required is missing, It's the way it's documented all through the library